### PR TITLE
Implement storageloc 

### PR DESCRIPTION
**Pull request by leeyimin**
- [x] Apply singleton to Config
- [x] Implement StoragelocCommand
- [x] Update CommandParser
- [x] Write storageloc Test
#35

### CODE REVIEW

**Comment by leeyimin** [comment]
storageloc test fails during build but passes when ran by itself. Still in fix

File: src/main/java/seedu/taskman/commons/core/config/ConfigData.java [hunk 1]
```diff
@@ -61,17 +58,30 @@ public void setTaskManName(String taskManName) {
...
+        return clone;
+    }
 
     @Override
     public boolean equals(Object other) {
         if (other == this){
             return true;
         }
-        if (!(other instanceof Config)){ //this handles null as well.
+        if (!(other instanceof ConfigData)){ //this handles null as well.
```
Comment (kazuhiroinaba) [inline src/main/java/seedu/taskman/commons/core/config/ConfigData.java]: Add space after condition. ") {"

File: src/main/java/seedu/taskman/commons/util/FileUtil.java [hunk 2]
```diff
@@ -92,4 +92,14 @@ public static String getPath(String pathWithForwardSlash) {
             throws IOException {
         return JsonUtil.fromJsonString(FileUtil.readFromFile(jsonFile), classOfObjectToDeserialize);
     }
+
+    /**
+     *
+     * @param path
+     * @return absolute path of given path
+     */
+    public static String getAbsolutePath(String path){
```
Comment (kazuhiroinaba) [inline src/main/java/seedu/taskman/commons/util/FileUtil.java]: Add space at  ") {"

File: src/main/java/seedu/taskman/logic/commands/StoragelocCommand.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            + "Example: " + COMMAND_WORD
+            + " C:/Users/Owner/Desktop/new_tasks.xml";
+
+    public static final String MESSAGE_SUCCESS = "Save location successfully updated to %1$s";
+    public static final String MESSAGE_FAILURE = "Attempt to Save Location was unsuccessful.\n" +
+            "Save location reverted to %1$s";
+
+    private final String filePath;
+
+    public StoragelocCommand(String filePath){
```
Comment (kazuhiroinaba) [inline src/main/java/seedu/taskman/logic/commands/StoragelocCommand.java]: Here too

File: src/main/java/seedu/taskman/logic/parser/CommandParser.java [hunk 4]
```diff
@@ -306,4 +314,27 @@ private Command prepareList(String args) {
         }
     }
 
+    /**
+     * Parses arguments in the context of the storageloc command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareStorageloc(String args){
```
Comment (kazuhiroinaba) [inline src/main/java/seedu/taskman/logic/parser/CommandParser.java]: lol i have yet to add a better comment than space

File: src/test/java/seedu/taskman/storage/StorageManagerTest.java [hunk 5]
```diff
@@ -76,6 +76,13 @@ public void handleTaskManChangedEvent_exceptionThrown_eventRaised() throws IOExc
         assertTrue(eventCollector.get(0) instanceof DataSavingExceptionEvent);
     }
 
+    @Test
+    public void setTaskManFilePath(){
```
Comment (kazuhiroinaba) [inline src/test/java/seedu/taskman/storage/StorageManagerTest.java]: space

File: src/test/java/seedu/taskman/logic/LogicManagerTest.java [hunk 6]
```diff
@@ -466,6 +478,72 @@ public void execute_list_filter_keywords_with_tags() throws Exception{
     }
 
 
+    private void assert_storage_location(String inputCommand, String expectedFeedback, String expectedPath, boolean success) throws IOException, DataConversionException {
```
Comment (kazuhiroinaba) [inline src/test/java/seedu/taskman/logic/LogicManagerTest.java]: Not sure if the line is too long, you can break it like the 2nd method after this one.

File: src/test/java/seedu/taskman/logic/LogicManagerTest.java [hunk 7]
```diff
@@ -466,6 +478,72 @@ public void execute_list_filter_keywords_with_tags() throws Exception{
...
+        CommandResult result = logic.execute(inputCommand);
+        assertEquals(result.feedbackToUser, expectedFeedback);
+        if (success) {
+            assertEquals(model.getTaskMan(), new TaskMan(storage.readTaskMan().get()));
+            assertEquals(storage.getTaskManFilePath(), expectedPath);
+        }
+        assertEquals(Config.getInstance().getTaskManFilePath(), expectedPath);
+    }
+
+    private String getStoragelocFeedback(String path, boolean success){
```
Comment (kazuhiroinaba) [inline src/test/java/seedu/taskman/logic/LogicManagerTest.java]: space here too

File: src/test/java/seedu/taskman/commons/core/ConfigTest.java [hunk 8]
```diff
@@ -3,32 +3,184 @@
...
+    }
+
+    @Test
+    public void getInstance(){
+        assertNotNull(Config.getInstance());
+        assertTrue(Config.getInstance() == Config.getInstance());
+    }
+
+    @Test
+    public void getInstance_afterReset(){
```
Comment (kazuhiroinaba) [inline src/test/java/seedu/taskman/commons/core/ConfigTest.java]: space here and up

File: src/test/java/seedu/taskman/commons/core/ConfigTest.java [hunk 9]
```diff
@@ -3,32 +3,184 @@
...
+        config.setLogLevel(Level.FINE);
+        originalData = config.getDataClone();
+        Config.setConfigFile(configFilePath);
+        Config.save();
+        config.readConfig(configFilePath);
+        assertEquals(config, originalData);
+    }
+
+    @Test
+    public void getInstance(){
```
Comment (inaba1231) [inline src/test/java/seedu/taskman/commons/core/ConfigTest.java]: space

File: src/test/java/seedu/taskman/commons/core/ConfigTest.java [hunk 10]
```diff
@@ -3,32 +3,184 @@
...
+    private static String TEST_DATA_FOLDER = FileUtil.getPath("./src/test/data/ConfigTest/");
+
     @Rule
     public ExpectedException thrown = ExpectedException.none();
 
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+    @Test
+    public void resetInstance(){
```
Comment (kazuhiroinaba) [inline src/test/java/seedu/taskman/commons/core/ConfigTest.java]: space

File: src/main/java/seedu/taskman/storage/StorageManager.java [hunk 11]
```diff
@@ -75,6 +75,11 @@ public void saveTaskMan(ReadOnlyTaskMan taskMan, String filePath) throws IOExcep
         taskManStorage.saveTaskMan(taskMan, filePath);
     }
 
+    @Override
+    public void setTaskManFilePath(String filePath){
```
Comment (kazuhiroinaba) [inline src/main/java/seedu/taskman/storage/StorageManager.java]: space

**Pull request merged**